packages/test/mocha-test-setup/src/mochaHooks.ts

```diff
@@ -24,112 +24,119 @@ const propsDict =
 		: undefined;
 
 const _global: any = global;
-class TestLogger implements ITelemetryBufferedLogger {
+
+/**
+ * A logger that adds context about the current test run to all events logged through it, like the test variant being
+ * run (odsp, r11s, etc) and the name of the current test (while in the context of a particular test running).
+ */
+class FluidTestRunLogger implements ITelemetryBufferedLogger {
+	private currentTestName: string | undefined;
+
 	send(event: ITelemetryBaseEvent) {
 		// TODO: Remove when issue #7061 is resolved.
 		// Don't log this event as we generate too much.
 		if (event.eventName === "fluid:telemetry:RouterliciousDriver:readBlob_end") {
 			return;
 		}
 
-		// The test logger is currently instantiated once and for each event triggered between begin and
-		// end of a test, in case the testName is undefined, we will use the currentTestName.
-		event.testName = this.testName ?? currentTestName;
+		if (this.currentTestName !== undefined) {
+			event.testName = this.currentTestName;
+		}
 		event.testVariant = testVariant;
-		event.hostName = pkgName;
 		this.parentLogger.send({
 			...event,
-			// If there's an override for the hostName in FLUID_LOGGER_PROPS,
-			// display that in the telemetry instead of the package name.
+			// Setting hostname to pkgName is the behavior we had for a long time, so keeping it just in case.
+			// But prefer a value set through FLUID_LOGGER_PROPS if it exists.
 			hostName: propsDict?.hostName ?? pkgName,
 			details: JSON.stringify(propsDict),
 		});
 	}
 	async flush() {
 		return this.parentLogger.flush();
 	}
-	constructor(
-		private readonly parentLogger: ITelemetryBufferedLogger,
-		private readonly testName: string | undefined,
-	) {}
+	constructor(private readonly parentLogger: ITelemetryBufferedLogger) {}
+
+	/**
+	 * Sets the test that is currently running.
+	 * The test name will be included in all events logged through the logger until {@link clearCurentTest} is called.
+	 * @param testName - The name of the test that is currently running.
+	 */
+	public setCurrentTest(testName: string) {
+		this.currentTestName = testName;
+	}
+
+	/**
+	 * Clears the test that is currently running.
+	 * Must be called after a given test has completed so that events logged outside the context of a test
+	 * don't include the name of the last test that ran.
+	 */
+	public clearCurentTest() {
+		this.currentTestName = undefined;
+	}
 }
 const nullLogger: ITelemetryBufferedLogger = {
 	send: () => {},
 	flush: async () => {},
 };
 
+// Keep references to the original console functions so we can restore them after each test.
 const log = console.log;
 const error = console.log;
 const warn = console.warn;
-let currentTestLogger: ITelemetryBufferedLogger | undefined;
-let currentTestName: string | undefined;
-let originalLogger: ITelemetryBufferedLogger;
+
+const originalLogger = _global.getTestLogger?.() ?? nullLogger;
+const testLogger = new FluidTestRunLogger(originalLogger);
```

Comment:
I think it's fine to do this when the module loads instead of inside `beforeAll`? I deliberately kept the bit that redefines `getTestLogger` inside `beforeAll` so that doesn't happen unless we're actually running tests, but getting the logger and creating its wrapper seems fine here. But please poke holes into this if you can see any.

Reply:
ehh it gives me mild pause, since putting this here puts dependencies on module load order. I'd at least appreciate a comment justifying why it's ok to do this, since the failure mode if something goes wrong in the future is pretty nasty (don't think we have any alerts for "where'd our telemetry go?").

If there isn't a pretty convincing reason why we're guaranteed this to be OK, I'd somewhat prefer something like the previous setup, perhaps with a helper function to grab the logger + assert it has been initialized.

Reply:
Moved back to that pattern.